store/bench_test.go

```diff
@@ -1,35 +1,313 @@
 package store
 
 import (
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 
+	dbm "github.com/cometbft/cometbft-db"
 	"github.com/cometbft/cometbft/internal/test"
-	"github.com/cometbft/cometbft/types"
-	cmttime "github.com/cometbft/cometbft/types/time"
+	sm "github.com/cometbft/cometbft/state"
 )
 
-// TestLoadBlockExtendedCommit tests loading the extended commit for a previously
-// saved block. The load method should return nil when only a commit was saved and
-// return the extended commit otherwise.
-func BenchmarkRepeatedLoadSeenCommitSameBlock(b *testing.B) {
-	state, bs, cleanup := makeStateAndBlockStore()
-	defer cleanup()
-	h := bs.Height() + 1
-	block := state.MakeBlock(h, types.MakeData(test.MakeNTxs(h, 10)), new(types.Commit), nil, state.Validators.GetProposer().Address)
-	seenCommit := makeTestExtCommitWithNumSigs(block.Header.Height, cmttime.Now(), 100).ToCommit()
-	ps, err := block.MakePartSet(types.BlockPartSizeBytes)
+// setupBlockStore creates a new block store for benchmarking
+func setupBlockStore(b *testing.B, storeType string) (sm.State, interface{}, func()) {
+	config := test.ResetTestRoot("block_store_bench")
+	stateStore := sm.NewStore(dbm.NewMemDB(), sm.StoreOptions{
```

Comment:
For this test file based DB should be used, otherwise we're comparing saving to files vs saving to memory.